### PR TITLE
PrerequisiteWAO, XORMinMaxWAO, Wizardlet.load, FObjectHolder + FObjectView Fixes

### PR DESCRIPTION
**Pull request by tharmaman**
- Added a PrerequisiteWAO which allows a wizardlet to load data from a direct prerequisite
- Added an XORMinMaxWAO which allows a wizardlet to load data from the object selected by a direct XORMinMax
- Added a preSet to wizardletIndex which calls load before a wizardlet changes to a new one.
- Added an FObjectHolder which is a a simple FObjectClass with a single FObjectProperty in order to set this to Capability.of where we don't know specifically what the data will be
- Fixed the choiceView used for FObjectView.objectClass which would choke and render undefined

### CODE REVIEW

File: src/foam/u2/wizard/PrerequisiteWAO.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        const currentWizardlet = wizardlet.prerequisiteWizardlets[i];
+
+        if ( 
+          currentWizardlet.capability && 
+          currentWizardlet.capability.id == this.prerequisiteCapabilityId 
+        ){
+          prereqWizardlet = currentWizardlet;
+          break;
+        }
+      }
```
Comment (KernelDeimos) [inline src/foam/u2/wizard/PrerequisiteWAO.js]: This entire block can be replaced with

```javascript
let prereqWizardlet = wizardlet.prerequisiteWizardlets.find(w =>
  w.capability && w.capability.id == this.prerequisiteCapabilityId)
```

File: src/foam/u2/wizard/PrerequisiteWAO.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            `prerequisiteCapabilityId: ${this.prerequisiteCapabilityId}'s data does not have the property ${this.propertyName}`
+          );
+        }
+
+        clonedPrereqWizardletData = prereqWizardletData[this.propertyName].clone();
+      } else {
+        clonedPrereqWizardletData = prereqWizardletData.clone();
+      }
+
+      var FObjectHolder = this.FObjectHolder.create({ fobject: clonedPrereqWizardletData });
```
Comment (KernelDeimos) [inline src/foam/u2/wizard/PrerequisiteWAO.js]: Variable name should be `fObjectHolder` - this casing implies the variable stores a class.

File: src/foam/u2/wizard/XORMinMaxWAO.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        var currentWizardlet = wizardlet.prerequisiteWizardlets[i];
+
+        if ( 
+          currentWizardlet.capability && 
+          currentWizardlet.capability.id == this.minMaxCapabilityId 
+        ){
+          prereqMinMaxWizardlet = currentWizardlet;
+          break;
+        }
+      }
```
Comment (KernelDeimos) [inline src/foam/u2/wizard/XORMinMaxWAO.js]: Similar comment as `PrerequisiteWAO`

File: src/foam/u2/wizard/XORMinMaxWAO.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      for ( var i = 0; i < prereqMinMaxWizardlet.prerequisiteWizardlets.length; i++ ){
+        var currentWizardlet = prereqMinMaxWizardlet.prerequisiteWizardlets[i];
+
+        if ( 
+          currentWizardlet.capability && 
+          currentWizardlet.capability.id == selectedCapabilityId 
+        ){
+          var clonedWizardletData = currentWizardlet.data.clone();
+
+          var FObjectHolder = this.FObjectHolder.create({ fobject: clonedWizardletData });
```
Comment (KernelDeimos) [inline src/foam/u2/wizard/XORMinMaxWAO.js]: Similar comment as `PrerequisiteWAO`

**Review by sarthak-marwaha — COMMENTED** [review]
FObjectView changes LGTM

File: src/foam/u2/wizard/FObjectHolder.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+ * The intellectual and technical concepts contained
+ * herein are proprietary to nanopay Corporation
+ * and may be covered by Canadian and Foreign Patents, patents
+ * in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from nanopay Corporation.
+ */
+
+ foam.CLASS({
```
Comment (KernelDeimos) [inline src/foam/u2/wizard/FObjectHolder.js]:
```suggestion
foam.CLASS({
```

File: src/foam/u2/wizard/PrerequisiteWAO.js (hunk 1 above)
Comment (tharmaman) [reply src/foam/u2/wizard/PrerequisiteWAO.js]: Done.

File: src/foam/u2/wizard/PrerequisiteWAO.js (hunk 2 above)
Comment (tharmaman) [reply src/foam/u2/wizard/PrerequisiteWAO.js]: Done.

File: src/foam/u2/wizard/XORMinMaxWAO.js (hunk 3 above)
Comment (tharmaman) [reply src/foam/u2/wizard/XORMinMaxWAO.js]: Done.

File: src/foam/u2/wizard/XORMinMaxWAO.js (hunk 4 above)
Comment (tharmaman) [reply src/foam/u2/wizard/XORMinMaxWAO.js]: Done.

**Review by nanopay-arthur — APPROVED** [review]

File: src/foam/u2/wizard/PrerequisiteWAO.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.u2.wizard',
+  name: 'PrerequisiteWAO',
+  implements: [ 'foam.u2.wizard.WAO' ],
+  flags: ['web'],
+  extends: 'foam.u2.wizard.ProxyWAO',
+  todo: [
```
Comment (KernelDeimos) [inline src/foam/u2/wizard/PrerequisiteWAO.js]: Kevin informed me that FOAM1 supported an `issues` axiom. We should use that convention here.
```suggestion
  issues: [
```

**Review by jlhughes — APPROVED** [review]

**Pull request merged**